homeassistant/components/powerwall/sensor.py

```diff
@@ -112,6 +113,116 @@ def _get_meter_average_voltage(meter: MeterResponse) -> float:
 )
 
 
+def _get_battery_charge(battery_data: BatteryResponse) -> float:
+    """Get the current value in %."""
+    ratio = float(battery_data.energy_remaining) / float(battery_data.capacity)
```

Comment:
We don't allow calculating state of entities in integrations that integrate devices or services. Please remove this sensor.

Reply:
For sure!   What is the alternative to provide the same information to the user?

Reply:
The other option is to have the underlying library return the value so we don't do the calculation in HA